addons/sale/models/sale_order_line.py

```diff
@@ -1347,6 +1348,15 @@ def _prepare_invoice_line(self, **optional_values):
         """
         self.ensure_one()
 
+        if self.product_id.type == 'combo':
+            qty_to_invoice = self.qty_to_invoice
+            with contextlib.suppress(ValueError):
+                qty_to_invoice = int(self.qty_to_invoice)
```

Comment:
@antoine162 Why does it need to be integer?  And I do not see the case where it would fail as int(6.5) e.g. returns 6.  Otherwise, it seems better this way.

Reply:
Ah, I think you're right :sweat_smile:

Indeed I don't see how the conversion would fail, and what I wanted to do is to convert the float to an integer only if it is a whole number, so I've changed it to actually do that :laughing: